lib/rubocop/cop/lint/require_relative_self_path.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Checks for uses a file requiring itself with `require_relative`.
+      #
+      # @example
+      #
+      #   # bad
+      #
+      #   # foo.rb
+      #   require_relative 'foo'
+      #   require_relative 'bar'
```

Comment:
```suggestion

```
I would suggest removing this because `bar` does not raise offense, only `foo` does.

Reply:
It would not be correct for `foo` to be `bar`.  I'd like to keep this for contrasted with the good case.